pkgs/development/compilers/ghc/8.10.4.nix

```diff
@@ -122,7 +124,7 @@ let
   # Use gold either following the default, or to avoid the BFD linker due to some bugs / perf issues.
   # But we cannot avoid BFD when using musl libc due to https://sourceware.org/bugzilla/show_bug.cgi?id=23856
   # see #84670 and #49071 for more background.
-  useLdGold = targetPlatform.linker == "gold" || (targetPlatform.linker == "bfd" && !targetPlatform.isMusl);
+  useLdGold = targetPlatform.linker == "gold" || (targetPlatform.linker == "bfd" && !targetPlatform.isMusl && !targetPlatform.isWindows);
```

Comment:
Does `ld.gold` not support windows? Noting something on this in a comment would be great.

Reply:
I'm not sure. 

```
Configuring ghc-prim-0.6.1...
ghc-cabal: Cannot find the program 'ld'. User-specified path
'/nix/store/yacysa24iax63f6m3wcm1y3jbw181v81-x86_64-w64-mingw32-binutils-wrapper-2.35.1/bin/x86_64-w64-mingw32-ld.gold'
does not refer to an executable and the program is not on the system path.
```

It doesn't appear to be currently built for windows
```
$ ls /nix/store/yacysa24iax63f6m3wcm1y3jbw181v81-x86_64-w64-mingw32-binutils-wrapper-2.35.1/bin/
x86_64-w64-mingw32-as  x86_64-w64-mingw32-ld  x86_64-w64-mingw32-ld.bfd
```

I'm a bit confused on where `bfd` is even coming from if all of `binutils` is disabled for windows?
https://github.com/NixOS/nixpkgs/blob/3e593430e07f36e0edce525c256b20f27934ef3f/pkgs/development/tools/misc/binutils/default.nix#L180

Maybe it's somehow enabled elsewhere but with `gold = false`  ?
https://github.com/NixOS/nixpkgs/blob/3e593430e07f36e0edce525c256b20f27934ef3f/pkgs/development/tools/misc/binutils/default.nix#L7

Reply:
That is the platform binutils runs on, not the one it targets.

Reply:
gold is ELF only so this is likely the issue. I think we maybe should make this fact discoverable via a passthru attribute or something in the binutils derivation? I think executable format is part of the platform set already, so shouldn't be too hard.

Reply:
Tried this in # #132538.

Reply:
> Tried this in # #132538.

Much better. I'll rebase this once that one's merged